### PR TITLE
new-logic-nuke-time

### PR DESCRIPTION
**Pull request by Gianfranco99**
# Update Dojo and Cairo versions, fix time-to-nuke calculation

This PR updates the development toolchain versions and fixes the time-to-nuke calculation logic in the taxes component.

Key changes:
- Update Dojo to 1.7.0-alpha.2, Cairo to 2.12.1, and Scarb to 2.12.1
- Add Sozo 1.7.0-alpha.2 to the tool versions
- Completely rewrite the `get_time_to_nuke` function to fix calculation issues:
  - Now properly calculates current unclaimed taxes from all neighbors
  - Uses the same tax calculation method as verification to ensure consistency
  - Correctly determines remaining time until nuke based on deficit between stake and unclaimed taxes
- Update tests to accommodate the new calculation logic with more realistic timestamps

The new implementation ensures more accurate nuke time predictions and fixes edge cases in the previous implementation.

### CODE REVIEW

**Comment by Gianfranco99** [comment]
* **#1299** <a href="https://app.graphite.com/github/pr/RuneLabsxyz/PonziLand/1299?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/RuneLabsxyz/PonziLand/1299?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>Merge queue</kbd> - adds this PR to the back of the merge queue
- <kbd>Hotfix 🔥</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/RuneLabsxyz?ref=merge-queue-instructions-in-stack-comment&prId=1299).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: contracts/src/tests/actions.cairo [hunk 1]
```diff
@@ -1267,18 +1272,17 @@ fn test_time_to_nuke() {
...
-    set_block_timestamp(time_to_nuke - block_timestamp);
+    set_block_timestamp(block_timestamp + time_to_nuke - 5);
+    let new_block_timestamp = get_block_timestamp();
+    let future_auction_price = get_suggested_sell_price(store, CENTER_LOCATION);
     let unclaimed_taxes = actions_system.get_unclaimed_taxes_per_neighbors_total(CENTER_LOCATION);
     assert!(unclaimed_taxes < land_stake.amount, "stake should be > unclaimed taxes");
     assert!(time_to_nuke != get_block_timestamp(), "should be not nukable yet");
-    set_block_timestamp(time_to_nuke);
-
+    set_block_timestamp(block_timestamp + time_to_nuke);
```
[the remark this answers is not in the document]
Comment (knownasred) [reply contracts/src/tests/actions.cairo]: In that case, would it cause issues to change this to something clearer? I thought block_timestamp is a public (shared) variable that increases as you set the block timestamp '^^

**Review by knownasred — APPROVED** [review]
